src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -585,6 +585,30 @@ ErrorIfOptionListHasNoTableName(List *optionList)
 }
 
 
+/*
+ * ForeignTableDropsTableNameOption returns true if given option list contains
+ * (DROP table_name).
+ */
+bool
+ForeignTableDropsTableNameOption(List *optionList)
+{
+	char *table_nameString = "table_name";
+	DefElem *option = NULL;
+	foreach_ptr(option, optionList)
+	{
+		char *optionName = option->defname;
+		DefElemAction optionAction = option->defaction;
+		if (strcmp(optionName, table_nameString) == 0 &&
```

Comment:
do we need case insensitive check? Can you maybe add a regression test like `tablE_Name 'table_name_drop'` or such?